tests/temp_tests/test_settings/test_localLauncher.py

```diff
@@ -0,0 +1,103 @@
+    assert localLauncher.launch_args._launch_args[key0] == val0
+    assert localLauncher.launch_args._launch_args[key1] == val1
+    assert localLauncher.launch_args._launch_args[key2] == val2
+
+    # Update our common run arguments
+    val2_upd = f"not-{val2}"
+    default_launcher_args[key2] = val2_upd
+
+    # Confirm previously created run settings are not changed
+    assert localLauncher.launch_args._launch_args[key2] == val2
+
+@pytest.mark.parametrize(
+    "env_vars",
+    [
+        pytest.param({}, id="no env vars"),
+        pytest.param({"env1": "abc"}, id="normal var"),
+        pytest.param({"env1": "abc,def"}, id="compound var"),
+        pytest.param({"env1": "xyz", "env2": "pqr"}, id="multiple env vars"),
+    ],
+)
+def test_update_env(env_vars):
+    """Ensure non-initialized env vars update correctly"""
+    localLauncher = LaunchSettings(launcher=LauncherType.Local)
+    localLauncher.update_env(env_vars)
+
+    assert len(localLauncher.env_vars) == len(env_vars.keys())
+
+def test_format_launch_args():
+    localLauncher = LaunchSettings(launcher=LauncherType.Local, launch_args={"-np": 2})
+    launch_args = localLauncher.format_launch_args()
+    assert launch_args == ["-np", "2"]
+
+@pytest.mark.parametrize(
+    "env_vars",
+    [
+        pytest.param({"env1": None}, id="null value not allowed"),
+        pytest.param({"env1": {"abc"}}, id="set value not allowed"),
+        pytest.param({"env1": {"abc": "def"}}, id="dict value not allowed"),
+    ],
+)
+def test_update_env_null_valued(env_vars):
+    """Ensure validation of env var in update"""
+    orig_env = {}
+
+    with pytest.raises(TypeError) as ex:
+        localLauncher = LaunchSettings(launcher=LauncherType.Local, env_vars=orig_env)
+        localLauncher.update_env(env_vars)
+
+@pytest.mark.parametrize(
+    "env_vars",
+    [
+        pytest.param({}, id="no env vars"),
+        pytest.param({"env1": "abc"}, id="normal var"),
+        pytest.param({"env1": "abc,def"}, id="compound var"),
+        pytest.param({"env1": "xyz", "env2": "pqr"}, id="multiple env vars"),
+    ],
+)
+def test_update_env_initialized(env_vars):
+    """Ensure update of initialized env vars does not overwrite"""
+    orig_env = {"key": "value"}
+    localLauncher = LaunchSettings(launcher=LauncherType.Local, env_vars=orig_env)
+    localLauncher.update_env(env_vars)
+
+    combined_keys = {k for k in env_vars.keys()}
+    combined_keys.update(k for k in orig_env.keys())
+
+    assert len(localLauncher.env_vars) == len(combined_keys)
+    assert {k for k in localLauncher.env_vars.keys()} == combined_keys
+
+def test_format_env_vars():
+    env_vars={
+            "A": "a",
+            "B": None,
+            "C": "",
+            "D": "12",
+        }
+    localLauncher = LaunchSettings(launcher=LauncherType.Local, env_vars=env_vars)
+    assert isinstance(localLauncher._arg_builder, LocalArgBuilder)
+    assert localLauncher.format_env_vars() == ["A=a", "B=", "C=", "D=12"]
```

Comment:
Not wrong, just a confirmation; is an env var dict `{"KEY": ""}` functionally equivalent to `{"KEY": None}`? It looks like that is what this test is testing for, but my intuition would be that `{"KEY": None}` would make sure that the `"KEY"` environment variable was unset from the executing environment. Might be worth getting more by in form the group on this issue, as I think we have been inconsistent in the past.